android/java/org/chromium/chrome/browser/sync/settings/BraveManageSyncSettings.java

```diff
@@ -54,11 +67,82 @@ public void onCreatePreferences(@Nullable Bundle savedInstanceState, String root
             syncAutofill.setTitle(R.string.brave_sync_autofill);
         }
 
+        assert mSyncEverything != null : "Something has changed in the upstream!";
+
         getPreferenceScreen().removePreference(mGoogleActivityControls);
         getPreferenceScreen().removePreference(mSyncEncryption);
 
         findPreference(PREF_ADVANCED_CATEGORY).setVisible(false);
 
         mSyncPaymentsIntegration.setVisible(false);
+
+        mPrefSyncPasswords = findPreference(PREF_SYNC_PASSWORDS);
+        assert mPrefSyncPasswords != null : "Something has changed in the upstream!";
+
+        overrideWithAuthConfirmationSyncPasswords();
+        overrideWithAuthConfirmationSyncEverything();
+    }
+
+    private void showScreenLockToast() {
+        Toast.makeText(ContextUtils.getApplicationContext(),
+                     R.string.password_sync_type_set_screen_lock, Toast.LENGTH_LONG)
+                .show();
+    }
+
+    private void overrideWithAuthConfirmationSyncPasswords() {
+        overrideWithAuthConfirmation(mPrefSyncPasswords);
+    }
+
+    private void overrideWithAuthConfirmationSyncEverything() {
+        overrideWithAuthConfirmation(mSyncEverything);
+    }
+
+    private void overrideWithAuthConfirmation(ChromeSwitchPreference control) {
+        Preference.OnPreferenceChangeListener origSyncListner =
+                control.getOnPreferenceChangeListener();
+
+        control.setOnPreferenceChangeListener((Preference preference, Object newValue) -> {
+            if ((Boolean) newValue) {
```

Comment:
I am sure it is boolean, because it is handler of `ChromeSwitchPreference` and it can me either true or false.

And, since we don't own ManageSyncSettings class and screen, I added assert at c130030f6a1277931c4a35d8e67416527b000ee4 , to see at least on our debug build if it will be changed somehow in the future.